### PR TITLE
Code clean - colliding variable names

### PR DESCRIPTION
**Pull request by baude**
Clean up variable names that collide with imported package names.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: cmd/podman/parse/net.go [hunk 1]
```diff
@@ -158,12 +158,12 @@ func parseEnvOrLabelFile(envOrLabel map[string]string, filename, configType stri
 
 // ValidURL checks a string urlStr is a url or not
 func ValidURL(urlStr string) error {
-	url, err := url.ParseRequestURI(urlStr)
+	parsedURL, err := url.ParseRequestURI(urlStr)
 	if err != nil {
-		return fmt.Errorf("invalid url %q: %w", urlStr, err)
+		return fmt.Errorf("invalid parsedURL %q: %w", urlStr, err)
```
Comment (giuseppe) [inline cmd/podman/parse/net.go:163]: the original error message seems better

File: cmd/podman/parse/net.go [hunk 2]
```diff
@@ -158,12 +158,12 @@ func parseEnvOrLabelFile(envOrLabel map[string]string, filename, configType stri
...
 	}
-	if url.Scheme == "" {
-		return fmt.Errorf("invalid url %q: missing scheme", urlStr)
+	if parsedURL.Scheme == "" {
+		return fmt.Errorf("invalid parsedURL %q: missing scheme", urlStr)
```
Comment (giuseppe) [inline cmd/podman/parse/net.go:166]: same here

File: cmd/podman/parse/net.go (hunk 1 above)
Comment (baude) [reply cmd/podman/parse/net.go:163]: thanks! i think this is over-aggressive IDE stuff ... I will revert the message

File: libpod/kube.go [hunk 3]
```diff
@@ -1390,17 +1390,17 @@ func removeUnderscores(s string) string {
 // and returns them as kube annotations
 func getAutoUpdateAnnotations(ctrName string, ctrLabels map[string]string, useLongAnnotations bool) map[string]string {
 	autoUpdateLabel := "io.containers.autoupdate"
-	annotations := make(map[string]string)
+	a := make(map[string]string)
```
Comment (jakecorrenti) [inline libpod/kube.go:1393]: I prefer the original here. Dealer's choice, though.

File: pkg/api/handlers/compat/networks.go [hunk 4]
```diff
@@ -282,17 +282,17 @@ func CreateNetwork(w http.ResponseWriter, r *http.Request) {
 				s.Gateway = gw
 			}
 			if len(conf.IPRange) > 0 {
-				_, net, err := net.ParseCIDR(conf.IPRange)
+				_, n, err := net.ParseCIDR(conf.IPRange)
```
Comment (jakecorrenti) [inline pkg/api/handlers/compat/networks.go:285]:
```suggestion
				_, ipNet, err := net.ParseCIDR(conf.IPRange)
```

File: pkg/api/handlers/libpod/manifests.go [hunk 5]
```diff
@@ -419,11 +419,11 @@ func ManifestPush(w http.ResponseWriter, r *http.Request) {
 	options.Writer = writer
 
 	pushCtx, pushCancel := context.WithCancel(r.Context())
-	var digest string
+	var d string
```
Comment (jakecorrenti) [inline pkg/api/handlers/libpod/manifests.go:422]: I think `digest` was better here.

File: pkg/domain/infra/abi/system.go [hunk 6]
```diff
@@ -395,7 +395,7 @@ func unshareEnv(graphroot, runroot string) []string {
 }
 
 func (ic *ContainerEngine) Unshare(ctx context.Context, args []string, options entities.SystemUnshareOptions) error {
-	unshare := func() error {
+	u := func() error {
```
Comment (jakecorrenti) [inline pkg/domain/infra/abi/system.go:398]: I think `unshare` was better, but `u` is also _ok_ here.

File: pkg/specgen/generate/kube/kube.go [hunk 7]
```diff
@@ -905,24 +905,24 @@ func k8sSecretFromSecretManager(name string, secretsManager *secrets.SecretsMana
 		return nil, err
 	}
 
-	var secrets map[string][]byte
-	if err := json.Unmarshal(inputSecret, &secrets); err != nil {
-		secrets = make(map[string][]byte)
+	var s map[string][]byte
```
Comment (jakecorrenti) [inline pkg/specgen/generate/kube/kube.go:908]: I think `secrets` is better here.

**Review by jakecorrenti — COMMENTED** [review]
some small naming nits

**Review by vrothberg — COMMENTED** [review]
Most of the changes rename variables.  Some look good to me, some not.  But all will make backports harder.  Is it worth it?

File: libpod/kube.go (hunk 3 above)
Comment (TomSweeneyRedHat) [reply libpod/kube.go:1393]: Except for iterators, I'm not at all a fan of single-character variable names.  When you're looking for the variable `a` in a 1400+ piece of code, good luck to the support engineer three years hence.

**Comment by baude** [comment]
the variable names collide with imported package names, this is consider a no-no in golang.  frankly, i was trying to reduce our footprint for  any sort of problem.

**Comment by baude** [comment]
@Luap99 ill give you final say here ... i think we should clean up this kind of stuff.  maybe it should be done on 4-5 and that is better timing.  if you want to break the tie here, then either suggest i finish it or close it.  no offense taken.

File: libpod/sqlite_state.go [hunk 8]
```diff
@@ -386,7 +386,7 @@ func (s *SQLiteState) ValidateDBConfig(runtime *Runtime) (defErr error) {
 		return nil
 	}
 
-	if err := checkField("os", os, runtimeOS); err != nil {
+	if err := checkField("dbOS", dbOS, runtimeOS); err != nil {
```
Comment (Luap99) [inline libpod/sqlite_state.go:389]: This string seems to change a error message and hints to a non existing db field now

File: pkg/systemd/quadlet/quadlet.go [hunk 9]
```diff
@@ -954,10 +954,10 @@ func ConvertVolume(volume *parser.UnitFile, name string, names map[string]string
 			opts.WriteString(fmt.Sprintf("gid=%d", gid))
 		}
 
-		copy, ok := volume.LookupBoolean(VolumeGroup, KeyCopy)
+		cp, ok := volume.LookupBoolean(VolumeGroup, KeyCopy)
 		if ok {
-			if copy {
-				podman.add("--opt", "copy")
+			if cp {
+				podman.add("--opt", "cp")
```
Comment (Luap99) [inline pkg/systemd/quadlet/quadlet.go:960]: this is not correct

**Review by Luap99 — COMMENTED** [review]
This seems to also change error handling with errors.Is() which is good but just not mentioned in the commit at all. Also what I do not like are the one later names that mean close to nothing now.

Did you search/replace the var names? The fact the it changed the word inside strings is concerning as it should only rename variables and not actual values.

In general I have no problem with this kind of change but this will never last. We already make extensive us of golangci-lint so you should figure out which checks to turn on so we can validate this in our CI. Relying on local reporting is nice but not sustainable for others. If we agree this is bad style then we need to enforce that with the linter for each PR.

**Pull request closed without merging**